romancal/patch_match/patch_match.py

```diff
@@ -0,0 +1,291 @@
+                bbintervals = iwcs.bounding_box.intervals
+                # This compensates for the half pixel adjustment in the general code.
+                image_shape = (bbintervals[0].upper + 0.5, bbintervals[1].upper + 0.5)
+            elif hasattr(iwcs, "pixel_shape") and iwcs.pixel_shape is not None:
+                image_shape = iwcs.pixel_shape
+        # Compute the image corners ra, dec from the wcs
+        (cxm, cxp), (cym, cyp) = (
+            (-0.5, image_shape[1] - 0.5),
+            (-0.5, image_shape[0] - 0.5),
+        )
+        image_corners = (iwcs(cxp, cyp), iwcs(cxm, cyp), iwcs(cxm, cym), iwcs(cxp, cym))
+    ptab = PATCH_TABLE
+    ra = ptab[:]["ra_center"]
+    dec = ptab[:]["dec_center"]
+    # # Convert all celestial coordinates to cartesion coordinates.
+    vec_centers = np.array(sgv.lonlat_to_vector(ra, dec)).transpose()
+    # # Organize corners into two ra, dec lists
+    # imra = np.array([point[0] for point in image_corners])
+    # imdec = np.array([point[1] for point in image_corners])
+    vec_im_corners = image_coords_to_vec(image_corners)
+    # Approximate center of image by averaging corner vectors
+    im_center = normalize_vector(vec_im_corners.mean(axis=1))
+    # Compute difference vector between image center and patch centers
+    diff = vec_centers - im_center
+    dist = np.sqrt((diff**2).sum(axis=1)) * 180 / np.pi
+    match = np.where(dist < 0.5)
+    ncandidates = len(match[0])
+    # Now see which of these that are close actually overlap the supplied image.
+    # (Is it necessary to check that the corners are in a sensible order?)
+    mra1 = ptab[match]["ra_corn1"]
+    mra2 = ptab[match]["ra_corn2"]
+    mra3 = ptab[match]["ra_corn3"]
+    mra4 = ptab[match]["ra_corn4"]
+    mdec1 = ptab[match]["dec_corn1"]
+    mdec2 = ptab[match]["dec_corn2"]
+    mdec3 = ptab[match]["dec_corn3"]
+    mdec4 = ptab[match]["dec_corn4"]
+    mcenters = vec_centers[match]
+    mra = np.vstack([mra1, mra2, mra3, mra4, mra1])
+    mdec = np.vstack([mdec1, mdec2, mdec3, mdec4, mdec1])
+    points = np.array(sgv.lonlat_to_vector(mra, mdec))
+    # Create polygon for supplied image_corners
+    pvec_im_corners = np.zeros((5, 3))
+    pvec_im_corners[:4] = vec_im_corners.transpose()
+    pvec_im_corners[4] = pvec_im_corners[0]
+    impoly = sgp.SingleSphericalPolygon(pvec_im_corners, im_center)
+    realmatch = []
+    for i in range(ncandidates):
+        # print(i)
+        cellpoints = points[:, :, i].transpose()
+        cellcenter = mcenters[i]
+        cellpoly = sgp.SingleSphericalPolygon(cellpoints, cellcenter)
+        if impoly.intersects_poly(cellpoly):
+            # print(f"candidate {i} intersects")
+            realmatch.append(i)
+    return match[0][realmatch], match[0]
+
+
+def get_corners(patch):
+    """
+    Construct a the vertex coordinates for a patch from a patch definition suitable
+    for plotting the defined region (x coordinates, y coordinates).
```

Comment:
this is not clear at all. it seems that this function returns Cartesian coordinates of vertex vectors `(x, y, z)` but the comment talks about plotting "the defined region" (x coordinates, y coordinates). What happened to the `z` coordinate?

Also, I would have expected this function to return the value of the variable `corners` - not Cartesian coordinates. So, maybe rename it to `get_cartesian_corners`.

Reply:
I changed the x, y to ra, dec, and changed the function to the name suggested.